src/index.js

```diff
@@ -1,266 +1,272 @@
+    }
+
+    const extractedName = contentParts && contentParts.length >= 1 ?
+      contentParts[1] :
+      this.params.url.split('/').pop().split('?')[0];
+
+    return this.params.nameCallback(extractedName);
+  }
+
+  getContentTypeFromFileSignature (file) {
+    const signatures = Object.assign({}, fileSignatures, this.params.customFileSignatures);
+
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      const first4BytesOfFile = file.slice(0, 4);
+
+      reader.onloadend = (evt) => {
+        if (evt.target.readyState !== FileReader.DONE) {
+          return;
+        }
+        // Since an array buffer is just a generic way to represent a binary buffer
+        // we need to create a TypedArray, in this case an Uint8Array
+        const uint = new Uint8Array(evt.target.result);
+        const bytes = [];
+
+        uint.forEach((byte) => {
+          // transform every byte to hexadecimal
+          bytes.push(byte.toString(16));
+        });
+
+        const hex = bytes.join('').toUpperCase();
+        resolve(signatures[hex]);
+      };
+
+      reader.onerror = reject;
+
+      // read first 4 bytes of sliced file as an array buffer
+      reader.readAsArrayBuffer(first4BytesOfFile);
+    });
+  }
+
+  getContentTypeFromResponseHeader () {
+    return this.request.getResponseHeader('content-type');
+  }
+
+  getContentType (file) {
+    return new Promise(async (resolve) => {
+      const { contentTypeDetermination } = this.params;
+      const defaultContentType = 'application/octet-stream';
+      let headerContentType;
+      let signatureContentType;
+
+      if (contentTypeDetermination === 'header' || contentTypeDetermination === 'full') {
+        headerContentType = this.getContentTypeFromResponseHeader();
+      }
+
+      if (contentTypeDetermination === 'signature' || contentTypeDetermination === 'full') {
+        signatureContentType = await this.getContentTypeFromFileSignature(file);
+      }
+
+      if (contentTypeDetermination === 'header') {
+        resolve(headerContentType ?? defaultContentType);
+      } else if (contentTypeDetermination === 'signature') {
+        resolve(signatureContentType ?? defaultContentType);
+      } else if (contentTypeDetermination === 'full') {
+        resolve(signatureContentType ?? headerContentType ?? defaultContentType);
+      } else {
+        resolve(defaultContentType);
+      }
+    });
+  }
+
+  createLink () {
+    let link = document.createElement('a');
+
+    link.style.display = 'none';
+    return link;
+  }
+
+  clickLink () {
+    let event;
+
+    try {
+      event = new MouseEvent('click');
+    } catch (e) {
+      event = document.createEvent('MouseEvent');
+      event.initMouseEvent('click', true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
+    }
+
+    this.link.dispatchEvent(event);
+  }
+
+  async getFile (response, fileName) {

```

Comment:
I see that you've removed the promises and replaced them with async functions. And I agree they are more readable, which is why I've tried this in the beginning, too. However, I ran into the following issue when testing the build:

```> js-file-downloader@1.1.23 test /Users/mbp/Git/js-file-downloader
> mocha --require @babel/register --colors ./test/*.js

/Users/mbp/Git/js-file-downloader/dist/js-file-downloader.min.js:383
        value: (i = d(regeneratorRuntime.mark(function e(t, n) {
                      ^

ReferenceError: regeneratorRuntime is not defined
    at /Users/mbp/Git/js-file-downloader/dist/js-file-downloader.min.js:383:23
    at Module.call (/Users/mbp/Git/js-file-downloader/dist/js-file-downloader.min.js:446:6)
    at __webpack_require__ (webpack://jsFileDownloader/webpack/bootstrap:19:22)
    at webpack://jsFileDownloader/webpack/bootstrap:83:10
    at factory (/Users/mbp/Git/js-file-downloader/dist/js-file-downloader.min.js:55:4)
    at webpack://jsFileDownloader/webpack/universalModuleDefinition:3:20
    at Object.<anonymous> (webpack://jsFileDownloader/webpack/universalModuleDefinition:1:1)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Module._compile (/Users/mbp/Git/js-file-downloader/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Object.newLoader [as .js] (/Users/mbp/Git/js-file-downloader/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (/Users/mbp/Git/js-file-downloader/test/index.js:2:1)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Module._compile (/Users/mbp/Git/js-file-downloader/node_modules/pirates/lib/index.js:99:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Object.newLoader [as .js] (/Users/mbp/Git/js-file-downloader/node_modules/pirates/lib/index.js:104:7)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at /Users/mbp/Git/js-file-downloader/node_modules/mocha/lib/mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/Users/mbp/Git/js-file-downloader/node_modules/mocha/lib/mocha.js:247:14)
    at Mocha.run (/Users/mbp/Git/js-file-downloader/node_modules/mocha/lib/mocha.js:576:10)
    at Object.<anonymous> (/Users/mbp/Git/js-file-downloader/node_modules/mocha/bin/_mocha:637:18)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47
npm ERR! Test failed.  See above for more details.

Process finished with exit code 1
```

I can't really make sense out of this, but I know it has something to do with the `async` functions.

Reply:
mmmh let me check 🤔